### ISSUE TITLE
ESR 45 creates symlinked files in new profile dir

### ISSUE THREAD

**Issue by dmik** [bitwiseworks/mozilla-os2#203]
In particular, `POFILE/prefs.js` is created as a symlink to `D:/Coding/mozilla/master/browser/app/profile/prefs.js` which eventually leads to modification of the source (which is under Git control so it gives constant status changes). This is definitely wrong and needs to be fixed.

This may be related to some new mozbuild magic that now e.g. creates symlinks to header files when gathering them under `OUT/dist/include` rather than copies the originals over. This thingy is actually a good thing as it saves disk space and helps C++ code parsers like the one in Qt Creator see that it's actually the same file and open the original for modifications instead of its symlinked shadow. But there should be no symlinks in a profile dir.

**Comment by dmik** [comment]
Ok, I found out what's wrong. `nsToolkitProfileService::CreateProfileInternal` uses `nsIFile::CopyTo` when copying profile defaults from the `browser/defaults/profile` subdirectory which is apparently wrong. There is `nsIFile::CopyToFolllowingSymlinks` (a new method appeared in ESR 38) that should be obviously used instead as it files the purpose. However, this method is the same as `CopyTo` on OS/2 for now so even if we use it, it will still just copy. In need to do a real `CopyToFollowingSymlinks` impl first.

BTW on Unix `CopyToFollowingSymlinks` is also just an equivalent of `CopyTo` (have no idea why). Perhaps, they just didn't hit this usage scenario.

I still couldn't find the exact reason why it never symlinked prefs.js on OS/2 back then but I'm not going to spend time on that as it's not important.

**Comment by dryeo** [comment]
Creating a new profile on SM does correctly create prefs.js though it 
does incorrectly create mimeTypes.rdf as a symlink. Never tried a new 
profile on TB.
The gdata-provider xpi is also incorrectly created with a bunch of 
symlinks under modules as well as defaults/preferences/preferences.js. 
All the other extensions seem to correctly copy rather then symlinking.
With 38ESR, Lightning didn't work due to an incorrect symlink in the 
calendar-timezones xpi IIRC, I just manually fixed it and uploaded the 
corrected symlink.

**Comment by dmik** [comment]
The reason why these files are symlinks in `dist/bin` is because they are listed in FINAL_TARGET_FILES in `moz.build` and files listed there are always symlinked if the target FS supports symlinks. As I said, I don't know why it didn't work on OS/2 in previous versions but that's irrelevant now as now it works "as designed". Also, the problem will go away after zipping the distro cause zip follows symlinks by default. But still, it needs to be fixed so that the development enviromnent works right. And the proper place for a fix is profile creation. It's too dangerous to support symlinks there as it's generally not predictable where they will point to after profile creation.

**Comment by dmik** [comment]
In fact, there are more places where symlinks are copied to the profile (e.g. `toolkit/mozapps/extensions/AddonManage.jsm` which copies `blocklist.xml`). So fixing `nsToolkitProfileService::CreateProfileInternal` (and `nsIFile::CopyToFollowingSymlinks`) would certainly be not enough at all (and I didn't feel like I wanted to find all these numerous places where stuff is copied from various parts to a new profile). So I found a better solution: I simply disabled symlinking for all files copied to `dist/bin` (as it doesn't make much sense). See the above commit.
